### DISCUSSION TITLE
non-diffracting beams

### DISCUSSION THREAD

**Discussion by kanikashh** [NanoComp/meep#2123] [Q&A]
Can I use meep for generating non-diffracting beams of various kinds and study their interaction with matter. I am new to numerical simulation techniques so I,m not able to find the right tool.

**Comment by oskooi** [comment]
It is possible to simulate non-diffracting beams using Meep although currently there is no built-in feature that supports this. A [Bessel beam](https://en.wikipedia.org/wiki/Bessel_beam) mode source would be a nice feature to have similar to a [Gaussian beam](https://meep.readthedocs.io/en/latest/Python_User_Interface/#gaussianbeamsource). Adding this feature would involve using the principle of equivalence which is described in [Section 4.2 of our book chapter](https://arxiv.org/abs/1301.5366). As an example, this [paper](https://opg.optica.org/oe/fulltext.cfm?uri=oe-26-22-28727&id=399326) describes an approach specific to FDTD.
